docs/src/prototypes/dropdowns/inputWithDropdown.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import keyboardKey from 'keyboard-key'
+import { Dropdown, DropdownProps } from '@stardust-ui/react'
+
+import { atMentionItems, AtMentionItem } from './dataMocks'
+import { insertTextAtCursorPosition } from './utils'
+import { CustomPortal } from './CustomPortal'
+
+interface InputWithDropdownState {
+  dropdownOpen?: boolean
+  searchQuery?: string
+}
+
+const editorStyle: React.CSSProperties = {
+  backgroundColor: 'lightgrey',
+  padding: '5px',
+  minHeight: '100px',
+  outline: 0,
+}
+
+class InputWithDropdown extends React.Component<{}, InputWithDropdownState> {
+  private readonly initialState: InputWithDropdownState = {
+    dropdownOpen: false,
+    searchQuery: '',
+  }
+
+  private contendEditableRef = React.createRef<HTMLDivElement>()
+
+  state = this.initialState
+
+  render() {
+    return (
+      <>
+        <div
+          contentEditable
+          ref={this.contendEditableRef}
+          onKeyUp={this.handleEditorKeyUp}
+          style={editorStyle}
+        />
+        {this.renderDropdown()}
```

Comment:
NIT: Do we really need `renderDropdown()`, it looks quite simple to be inlined

Reply:
with the latest refactoring..no, we don't need it, inlined it, thanks